### PR TITLE
Delete old records + contact data

### PR DESCRIPTION
**Pull request by insacuri**
# Motivation and Context
Previously all the data from every printfile generated was saved in the actionexporter database in perpetuity.  This data included names and email addresses of respondents.  As part of the GDPR work, we're going to remove the data that is no longer needed.  Every morning at 3am, the /delete-old-records endpoint will be hit and delete all data related to exportFiles that were successfully sent over 90 days ago.  This gives us a reasonable amount of time to replay anything that might need to be replayed, but anything longer then 90 days will almost certainly not be needed again.

Note:  The address data isn't deleted in this PR.  This is more difficult because of the Many to One relationship the addresses have with the action instructions (one address record is used for multiple action instructions) and the documentation isn't super clear how to clear down an address after the last action instruction has finished using it.

# What has changed

# How to test?
- Port forward to your environment
- Run this script [recreate_actionexporter.txt](https://github.com/ONSdigital/rm-actionexporter-service/files/5232758/recreate_actionexporter.txt) against your database to populate some test data
- `curl -X DELETE http://localhost:8141/delete-old-records`, then look at the logs to see one of the exportjobs deleted (and all its associated data)

# Links

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/action/export/scheduled/DeleteProcessor.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      // If the exportJob has any files unsent or younger then 90 days, we don't want to remove
+      // the linking id.
+      boolean allFilesFromExportJobSent = true;
+
+      for (ExportFile exportFile : exportFiles) {
+        log.info("Working on exportFile with id [" + exportFile.getId() + "]");
+        Timestamp dateSuccessfullySent = exportFile.getDateSuccessfullySent();
+        Date ninetyDaysAgo = new Date(System.currentTimeMillis() - (90 * DAY_IN_MS));
+
+        if (dateSuccessfullySent != null && dateSuccessfullySent.before(ninetyDaysAgo)) {
```
Comment (warrenbailey) [inline src/main/java/uk/gov/ons/ctp/response/action/export/scheduled/DeleteProcessor.java:56]: We may be able remove this check as I don't think the successfully sent flag is used and just delete all entries over 90 days

**Review by warrenbailey — APPROVED** [review]

File: src/main/java/uk/gov/ons/ctp/response/action/export/endpoint/DeleteEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  private static final Logger log = LoggerFactory.getLogger(DeleteEndpoint.class);
+
+  @Autowired private DeleteProcessor deleteProcessor;
+
+  @RequestMapping(method = RequestMethod.DELETE)
+  public ResponseEntity<String> triggerDelete() throws CTPException {
+    try {
+      deleteProcessor.triggerDelete();
+      return ResponseEntity.ok().body("Deletion of old records completed");
+    } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (insacuri) [reply src/main/java/uk/gov/ons/ctp/response/action/export/endpoint/DeleteEndpoint.java]: Done

File: src/main/java/uk/gov/ons/ctp/response/action/export/scheduled/DeleteProcessor.java (hunk 1 above)
Comment (insacuri) [reply src/main/java/uk/gov/ons/ctp/response/action/export/scheduled/DeleteProcessor.java:56]: I looked at the doco for this and it throws a NullPointerException if the field is null.  Is it okay for me to leave it as is?

**Pull request merged**